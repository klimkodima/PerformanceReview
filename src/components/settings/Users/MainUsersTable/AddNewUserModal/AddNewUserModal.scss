.add-new-user {
  min-width: 700px;
  margin-bottom: 20px;

  &__title {
    display: flex;
    justify-content: center;
    align-items: center;
    color: #6c6c6c;

    h5 {
      font-size: 22px;
      font-weight: 700;
    }
  }

  &__item {
    display: flex;
    align-items: center;
    margin-top: 20px;

    info-icon {
      position: absolute;
      top: 0;
      right: -30px;
    }

    .add-new-user-select {
      width: 300px;

      select {
        padding: 5px 5px;
        font-size: 18px;
        color: #787878;
        font-weight: 400;
      }
    }

    .placeholder > select {
      opacity: 0.5;
      font-size: 18px;
    }
  }

  label {
    margin-right: 40px;
    width: 100px;
  }

  .add-new-user-input {
    input {
      min-width: 400px;
      padding: 5px 10px;
    }
  }

  .preset-working-hours-checkbox {
    margin-left: 40px;
  }

  #preset-working-hours-label {
    margin: 0;
  }

  .add-new-user-wrapper {
    width: 800px;
  }
}
