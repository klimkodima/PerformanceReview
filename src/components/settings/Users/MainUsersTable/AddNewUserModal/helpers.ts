import { UserInfoType } from 'src/store/Users';
import { TeamsDataResponseType } from 'src/store/Filter/types';
import { TeamsDataType } from './types';

export const createTeams = (
  users: UserInfoType[],
  teams: TeamsDataResponseType[]
): TeamsDataType[] => {
  const uniqueTeams = Array.from(
    new Map(users.map((item) => [item.teamName, item])).values()
  );

  return teams.map(({ teamName, teamLeaderName }) => {
    const currentValues = uniqueTeams.find(
      (team) => team.teamName === teamName
    );
    return {
      teamName,
      teamLeaderName,
      teamId: currentValues?.teamId ? currentValues?.teamId : 0
    };
  });
};
