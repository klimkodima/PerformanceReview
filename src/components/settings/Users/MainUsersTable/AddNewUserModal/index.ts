export { default as AddNewUserModal } from './AddNewUserModalContainer';
