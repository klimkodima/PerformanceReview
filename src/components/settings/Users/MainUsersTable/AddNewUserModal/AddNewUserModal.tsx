import { ChangeEventHandler, memo, ReactElement } from 'react';

import { useTranslation } from 'react-i18next';
import InfoIcon from '@mui/icons-material/Info';
import { Checkbox, TextField } from '@mui/material';
import { SelectChangeEvent } from '@mui/material/Select';

import { ErrorTooltip, InfoTooltip } from 'src/components/common';
import LevelSelect from './LevelSelect';
import TeamSelect from './TeamSelect';
import RoleSelect from './RoleSelect';

import { AddNewUserFormValuesType } from 'src/store/Users';
import { AddNewUserValidationType, TeamsDataType } from './types';

import './AddNewUserModal.scss';

type AddNewUserModalPropsType = {
  onSelectChange: (label: string) => (e: SelectChangeEvent) => void;
  onInputChange: (
    label: string
  ) => ChangeEventHandler<HTMLInputElement | HTMLTextAreaElement>;
  formValues: AddNewUserFormValuesType;
  teams: TeamsDataType[];
  isShowTeamError: boolean;
  validation: AddNewUserValidationType;
  isShowLevelAndTeamInputs: boolean;
};

const AddNewUserModal = memo(
  ({
    onSelectChange,
    onInputChange,
    formValues,
    teams,
    isShowTeamError,
    validation,
    isShowLevelAndTeamInputs
  }: AddNewUserModalPropsType): ReactElement => {
    const { t } = useTranslation();
    return (
      <div className='add-new-user'>
        <div className='add-new-user__title'>
          <h5>{t('add_new_user_modal.title')}</h5>
        </div>
        <div className='add-new-user-wrapper'>
          <div className='add-new-user__item' data-testid='add-new-user__item'>
            <label id='new-user-username'>
              {t('add_new_user_modal.username_label')}
            </label>
            <TextField
              placeholder={t('add_new_user_modal.username_placeholder')}
              className='add-new-user-input'
              id='new-user-username'
              value={formValues.username}
              onChange={onInputChange('username')}
            />
            {validation.username && (
              <ErrorTooltip title={validation.username} />
            )}
          </div>
          <div className='add-new-user__item' data-testid='add-new-user__item'>
            <label id='email-login'>
              {t('add_new_user_modal.email_label')}
            </label>
            <TextField
              placeholder={t('add_new_user_modal.email_placeholder')}
              className='add-new-user-input'
              id='email-login'
              value={formValues.email}
              onChange={onInputChange('email')}
            />
            {validation.email && <ErrorTooltip title={validation.email} />}
          </div>
          <div className='add-new-user__item' data-testid='add-new-user__item'>
            <label id='new-user-password'>
              {t('add_new_user_modal.password_label')}
            </label>
            <TextField
              type='password'
              placeholder={t('add_new_user_modal.password_placeholder')}
              className='add-new-user-input'
              id='new-user-password'
              value={formValues.password}
              onChange={onInputChange('password')}
            />
            <InfoTooltip
              title={t('add_new_user_modal.password_info')}
              icon={
                <InfoIcon
                  className='info-icon'
                  color='primary'
                  data-testid='info-icon-test'
                />
              }
            />
            {validation.password && (
              <ErrorTooltip title={validation.password} />
            )}
          </div>
          <div className='add-new-user__item' data-testid='add-new-user__item'>
            <RoleSelect
              roleName={formValues.roleName}
              validation={validation.roleName}
              onSelectChange={onSelectChange}
            />
            <div className='preset-working-hours-checkbox'>
              <Checkbox
                id='preset-working-hours-label'
                value={formValues.thirdParty}
                onChange={onInputChange('thirdParty')}
              />
              <label id='preset-working-hours-label'>
                {t('add_new_user_modal.preset_working_hours_label')}
              </label>
              <InfoTooltip
                title={t('add_new_user_modal.preset_working_hours_info')}
                icon={
                  <InfoIcon
                    className='info-icon'
                    color='primary'
                    data-testid='info-icon-test'
                  />
                }
              />
            </div>
          </div>
          {isShowLevelAndTeamInputs && (
            <div>
              <div
                className='add-new-user__item'
                data-testid='add-new-user__item'
              >
                <TeamSelect
                  teamName={formValues.teamName ? formValues.teamName : ''}
                  teams={teams}
                  isShowTeamError={isShowTeamError}
                  validation={validation.teamName}
                  onSelectChange={onSelectChange}
                />
              </div>
              <div
                className='add-new-user__item'
                data-testid='add-new-user__item'
              >
                <LevelSelect
                  level={formValues.level ? formValues.level : ''}
                  roleName={formValues.roleName}
                  validation={validation.level}
                  onSelectChange={onSelectChange}
                />
              </div>
            </div>
          )}
        </div>
      </div>
    );
  }
);

export default AddNewUserModal;

AddNewUserModal.displayName = 'AddNewUserModal';
