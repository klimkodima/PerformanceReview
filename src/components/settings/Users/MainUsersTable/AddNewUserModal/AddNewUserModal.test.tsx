import renderer from 'react-test-renderer';
import { fireEvent, render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import { I18nextProvider } from 'react-i18next';

import i18n from 'src/i18n';
import store from 'src/store';
import AddNewUserModal from './AddNewUserModal';
import AddNewUserModalContainer from './AddNewUserModalContainer';

const mockTeams = [
  {
    teamName: 'X-Rays',
    teamLeaderName: 'Tatiana Klimchenya',
    teamId: 1
  },
  {
    teamName: 'Charlie',
    teamLeaderName: 'Anastasya Karman',
    teamId: 2
  },
  {
    teamName: 'Alpha',
    teamLeaderName: 'Bogdana Korniyuk',
    teamId: 3
  }
];

const mockValidation = {
  email: null,
  password: null,
  roleName: null,
  username: null,
  level: null,
  teamName: null
};

const mockFormValues = {
  fullName: '',
  email: '',
  roleName: '',
  password: '',
  username: '',
  thirdParty: false,
  level: '',
  teamName: ''
};

describe('AddNewUserModal test', () => {
  test('renders correctly', () => {
    const component = renderer
      .create(
        <Provider store={store}>
          <I18nextProvider i18n={i18n}>
            <AddNewUserModal
              isShowTeamError={false}
              validation={mockValidation}
              teams={mockTeams}
              isShowLevelAndTeamInputs={false}
              formValues={mockFormValues}
              onInputChange={() => (event) => event.target}
              onSelectChange={() => (event) => event.target}
            />
          </I18nextProvider>
        </Provider>
      )
      .toJSON();

    expect(component).toMatchSnapshot();
  });
});

describe('AddNewUserModalContainer test', () => {
  test('renders correctly', () => {
    const root = document.createElement('div');
    root.setAttribute('id', 'portal');
    document.body.appendChild(root);

    render(
      <Provider store={store}>
        <I18nextProvider i18n={i18n}>
          <AddNewUserModalContainer onCloseModalClick={() => true} />
        </I18nextProvider>
      </Provider>
    );

    expect(screen.getByTestId('modal-overlay')).toMatchSnapshot();
  });

  test('close modal', () => {
    const root = document.createElement('div');
    root.setAttribute('id', 'portal');
    document.body.appendChild(root);

    const cancelMock = jest.fn();

    render(
      <Provider store={store}>
        <I18nextProvider i18n={i18n}>
          <AddNewUserModalContainer onCloseModalClick={cancelMock} />
        </I18nextProvider>
      </Provider>
    );

    fireEvent.click(screen.getByText(/Cancel/i));

    expect(cancelMock).toBeCalled();
  });

  test('validation works', () => {
    const root = document.createElement('div');
    root.setAttribute('id', 'portal');
    document.body.appendChild(root);

    render(
      <Provider store={store}>
        <I18nextProvider i18n={i18n}>
          <AddNewUserModalContainer onCloseModalClick={() => true} />
        </I18nextProvider>
      </Provider>
    );

    const usernameInput = screen.getByPlaceholderText('Enter user name');
    const loginInput = screen.getByPlaceholderText('Enter user login');
    const passwordInput = screen.getByPlaceholderText('Enter user password');
    const roleSelect = screen.getByTestId('role-select');
    const saveBtn = screen.getByText('Add User');

    expect(saveBtn).toHaveClass('Mui-disabled');

    fireEvent.change(usernameInput, {
      target: { value: 'user' }
    });

    fireEvent.change(loginInput, {
      target: { value: 'login' }
    });

    fireEvent.change(passwordInput, {
      target: { value: 'pass' }
    });

    fireEvent.change(roleSelect.childNodes[0], {
      target: { value: 'ADMIN' }
    });
    expect(usernameInput).toHaveAttribute('value', 'user');
    expect(loginInput).toHaveAttribute('value', 'login');
    expect(saveBtn).not.toHaveClass('Mui-disabled');

    fireEvent.click(saveBtn);
    expect(saveBtn).toHaveClass('Mui-disabled');
    expect(screen.getAllByTestId('PriorityHighIcon').length).toBe(3);
  });

  test('show two more fields if selected role auditor or team leader', () => {
    const root = document.createElement('div');
    root.setAttribute('id', 'portal');
    document.body.appendChild(root);

    render(
      <Provider store={store}>
        <I18nextProvider i18n={i18n}>
          <AddNewUserModalContainer onCloseModalClick={() => true} />
        </I18nextProvider>
      </Provider>
    );

    const roleSelect = screen.getByTestId('role-select');
    expect(screen.getAllByTestId('add-new-user__item').length).toBe(4);

    fireEvent.change(roleSelect.childNodes[0], {
      target: { value: 'AUDITOR' }
    });

    expect(screen.getAllByTestId('add-new-user__item').length).toBe(6);
  });

  test('if validated field entered again, remove error notification', () => {
    const root = document.createElement('div');
    root.setAttribute('id', 'portal');
    document.body.appendChild(root);

    render(
      <Provider store={store}>
        <I18nextProvider i18n={i18n}>
          <AddNewUserModalContainer onCloseModalClick={() => true} />
        </I18nextProvider>
      </Provider>
    );

    const usernameInput = screen.getByPlaceholderText('Enter user name');
    const loginInput = screen.getByPlaceholderText('Enter user login');
    const passwordInput = screen.getByPlaceholderText('Enter user password');
    const roleSelect = screen.getByTestId('role-select');
    const saveBtn = screen.getByText('Add User');

    expect(saveBtn).toHaveClass('Mui-disabled');

    fireEvent.change(usernameInput, {
      target: { value: 'user' }
    });

    fireEvent.change(loginInput, {
      target: { value: 'login' }
    });

    fireEvent.change(passwordInput, {
      target: { value: 'pass' }
    });

    fireEvent.change(roleSelect.childNodes[0], {
      target: { value: 'ADMIN' }
    });
    expect(usernameInput).toHaveAttribute('value', 'user');
    expect(loginInput).toHaveAttribute('value', 'login');
    expect(saveBtn).not.toHaveClass('Mui-disabled');

    fireEvent.click(saveBtn);
    expect(saveBtn).toHaveClass('Mui-disabled');
    expect(screen.getAllByTestId('PriorityHighIcon').length).toBe(3);

    fireEvent.change(passwordInput, {
      target: { value: 'password' }
    });
    expect(screen.getAllByTestId('PriorityHighIcon').length).toBe(2);

    fireEvent.change(usernameInput, {
      target: { value: 'username1' }
    });
    expect(screen.getAllByTestId('PriorityHighIcon').length).toBe(1);

    fireEvent.change(loginInput, {
      target: { value: 'login@mail.ru' }
    });
    expect(screen.queryAllByTestId('PriorityHighIcon').length).toBe(0);
  });
});
