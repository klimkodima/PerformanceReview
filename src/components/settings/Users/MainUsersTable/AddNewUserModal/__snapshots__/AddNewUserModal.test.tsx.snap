// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddNewUserModal test renders correctly 1`] = `
<div
  className="add-new-user"
>
  <div
    className="add-new-user__title"
  >
    <h5>
      New User
    </h5>
  </div>
  <div
    className="add-new-user-wrapper"
  >
    <div
      className="add-new-user__item"
      data-testid="add-new-user__item"
    >
      <label
        id="new-user-username"
      >
        User Name:
      </label>
      <div
        className="MuiFormControl-root MuiTextField-root add-new-user-input css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
      >
        <div
          className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoFocus={false}
            className="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            disabled={false}
            id="new-user-username"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Enter user name"
            required={false}
            type="text"
            value=""
          />
          <fieldset
            aria-hidden={true}
            className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="css-hdw1oc"
            >
              <span
                className="notranslate"
              >
                ​
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
    <div
      className="add-new-user__item"
      data-testid="add-new-user__item"
    >
      <label
        id="email-login"
      >
        Email (login):
      </label>
      <div
        className="MuiFormControl-root MuiTextField-root add-new-user-input css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
      >
        <div
          className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoFocus={false}
            className="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            disabled={false}
            id="email-login"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Enter user login"
            required={false}
            type="text"
            value=""
          />
          <fieldset
            aria-hidden={true}
            className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="css-hdw1oc"
            >
              <span
                className="notranslate"
              >
                ​
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
    <div
      className="add-new-user__item"
      data-testid="add-new-user__item"
    >
      <label
        id="new-user-password"
      >
        Password:
      </label>
      <div
        className="MuiFormControl-root MuiTextField-root add-new-user-input css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
      >
        <div
          className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoFocus={false}
            className="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            disabled={false}
            id="new-user-password"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Enter user password"
            required={false}
            type="password"
            value=""
          />
          <fieldset
            aria-hidden={true}
            className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="css-hdw1oc"
            >
              <span
                className="notranslate"
              >
                ​
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <svg
        aria-hidden={true}
        aria-label="Length of password must be between 8 and 40."
        aria-labelledby={null}
        className="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeMedium info-icon css-ci5apu-MuiSvgIcon-root"
        data-mui-internal-clone-element={true}
        data-testid="info-icon-test"
        focusable="false"
        onBlur={[Function]}
        onFocus={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        viewBox="0 0 24 24"
      >
        <path
          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
        />
      </svg>
    </div>
    <div
      className="add-new-user__item"
      data-testid="add-new-user__item"
    >
      <label
        id="role-label"
      >
        Role:
      </label>
      <div
        className="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
      >
        <div
          className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl add-new-user-select placeholder css-1yk1gt9-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
          data-testid="role-select"
          onClick={[Function]}
        >
          <select
            aria-invalid={false}
            className="MuiNativeSelect-select MuiNativeSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-ffg8md-MuiNativeSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
            disabled={false}
            id="role-select"
            multiple={false}
            name="role-select"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={false}
            value=""
          >
            <option
              data-testid="role-select-ADMIN"
              hidden={false}
              value="ADMIN"
            >
              Admin
            </option>
            <option
              data-testid="role-select-TEAM_LEADER"
              hidden={false}
              value="TEAM_LEADER"
            >
              Team Leader
            </option>
            <option
              data-testid="role-select-AUDITOR"
              hidden={false}
              value="AUDITOR"
            >
              Auditor
            </option>
            <option
              data-testid="role-select-MANAGER"
              hidden={false}
              value="MANAGER"
            >
              Manager
            </option>
            <option
              data-testid="role-select-"
              hidden={true}
              value=""
            >
              Select user's role
            </option>
          </select>
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiNativeSelect-icon MuiNativeSelect-iconOutlined css-1g12qau-MuiSvgIcon-root-MuiNativeSelect-icon"
            data-testid="ArrowDropDownIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
          <fieldset
            aria-hidden={true}
            className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="css-hdw1oc"
            >
              <span
                className="notranslate"
              >
                ​
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        className="preset-working-hours-checkbox"
      >
        <span
          className="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
          onBlur={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={null}
        >
          <input
            className="PrivateSwitchBase-input css-1m9pwf3"
            data-indeterminate={false}
            id="preset-working-hours-label"
            onChange={[Function]}
            type="checkbox"
            value={false}
          />
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="CheckBoxOutlineBlankIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
            />
          </svg>
          <span
            className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </span>
        <label
          id="preset-working-hours-label"
        >
          Preset working hours schedule
        </label>
        <svg
          aria-hidden={true}
          aria-label="Set when the auditor/team lead uses the third-party time-management system instead of a corporate Jira project"
          aria-labelledby={null}
          className="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeMedium info-icon css-ci5apu-MuiSvgIcon-root"
          data-mui-internal-clone-element={true}
          data-testid="info-icon-test"
          focusable="false"
          onBlur={[Function]}
          onFocus={[Function]}
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
          viewBox="0 0 24 24"
        >
          <path
            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;

exports[`AddNewUserModalContainer test renders correctly 1`] = `
<div
  class="modal-overlay"
  data-testid="modal-overlay"
>
  <div
    class="modal-body "
  >
    <div
      class="add-new-user"
    >
      <div
        class="add-new-user__title"
      >
        <h5>
          New User
        </h5>
      </div>
      <div
        class="add-new-user-wrapper"
      >
        <div
          class="add-new-user__item"
          data-testid="add-new-user__item"
        >
          <label
            id="new-user-username"
          >
            User Name:
          </label>
          <div
            class="MuiFormControl-root MuiTextField-root add-new-user-input css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
          >
            <div
              class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                id="new-user-username"
                placeholder="Enter user name"
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-hdw1oc"
                >
                  <span
                    class="notranslate"
                  >
                    ​
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          class="add-new-user__item"
          data-testid="add-new-user__item"
        >
          <label
            id="email-login"
          >
            Email (login):
          </label>
          <div
            class="MuiFormControl-root MuiTextField-root add-new-user-input css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
          >
            <div
              class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                id="email-login"
                placeholder="Enter user login"
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-hdw1oc"
                >
                  <span
                    class="notranslate"
                  >
                    ​
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          class="add-new-user__item"
          data-testid="add-new-user__item"
        >
          <label
            id="new-user-password"
          >
            Password:
          </label>
          <div
            class="MuiFormControl-root MuiTextField-root add-new-user-input css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
          >
            <div
              class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                id="new-user-password"
                placeholder="Enter user password"
                type="password"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-hdw1oc"
                >
                  <span
                    class="notranslate"
                  >
                    ​
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <svg
            aria-hidden="true"
            aria-label="Length of password must be between 8 and 40."
            class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeMedium info-icon css-ci5apu-MuiSvgIcon-root"
            data-mui-internal-clone-element="true"
            data-testid="info-icon-test"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
            />
          </svg>
        </div>
        <div
          class="add-new-user__item"
          data-testid="add-new-user__item"
        >
          <label
            id="role-label"
          >
            Role:
          </label>
          <div
            class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
          >
            <div
              class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl add-new-user-select placeholder css-1yk1gt9-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
              data-testid="role-select"
            >
              <select
                aria-invalid="false"
                class="MuiNativeSelect-select MuiNativeSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-ffg8md-MuiNativeSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                id="role-select"
                name="role-select"
              >
                <option
                  data-testid="role-select-ADMIN"
                  value="ADMIN"
                >
                  Admin
                </option>
                <option
                  data-testid="role-select-TEAM_LEADER"
                  value="TEAM_LEADER"
                >
                  Team Leader
                </option>
                <option
                  data-testid="role-select-AUDITOR"
                  value="AUDITOR"
                >
                  Auditor
                </option>
                <option
                  data-testid="role-select-MANAGER"
                  value="MANAGER"
                >
                  Manager
                </option>
                <option
                  data-testid="role-select-"
                  hidden=""
                  value=""
                >
                  Select user's role
                </option>
              </select>
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiNativeSelect-icon MuiNativeSelect-iconOutlined css-1g12qau-MuiSvgIcon-root-MuiNativeSelect-icon"
                data-testid="ArrowDropDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-hdw1oc"
                >
                  <span
                    class="notranslate"
                  >
                    ​
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            class="preset-working-hours-checkbox"
          >
            <span
              class="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
            >
              <input
                class="PrivateSwitchBase-input css-1m9pwf3"
                data-indeterminate="false"
                id="preset-working-hours-label"
                type="checkbox"
                value="false"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="CheckBoxOutlineBlankIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </span>
            <label
              id="preset-working-hours-label"
            >
              Preset working hours schedule
            </label>
            <svg
              aria-hidden="true"
              aria-label="Set when the auditor/team lead uses the third-party time-management system instead of a corporate Jira project"
              class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeMedium info-icon css-ci5apu-MuiSvgIcon-root"
              data-mui-internal-clone-element="true"
              data-testid="info-icon-test"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
    <div
      class="modal-overlay__buttons"
    >
      <button
        class="MuiButton-root MuiButton-outlined MuiButton-outlinedSuccess MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root Mui-disabled  css-sxix9q-MuiButtonBase-root-MuiButton-root"
        disabled=""
        tabindex="-1"
        type="button"
      >
        <span
          class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="CheckIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M9 16.17 4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
            />
          </svg>
        </span>
        Add User
      </button>
      <button
        class="MuiButton-root MuiButton-outlined MuiButton-outlinedError MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root  css-180lf2n-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="CloseIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
            />
          </svg>
        </span>
        Cancel
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;
